File: 솝트 알고리즘 스터디/체육복.js
function solution(n, lost, reserve) {
    let losts = new Set(lost.filter(l => !reserve.includes(l)));
    let reserves = reserve.filter(r => !lost.includes(r));

    reserves.sort((a, b) => a - b);
    for (let person of reserves) {
        let left = person - 1;
        let right = person + 1;

        if (losts.has(left)) {
            losts.delete(left)
        } else if (losts.has(right)) {
            losts.delete(right)
        }
    }
    return n - [...losts].length;
}

File: 솝트 알고리즘 스터디/체육복.py
def solution(n, lost, reserve):

    lost, reserve = set(lost) - set(reserve), set(reserve) - set(lost)
    
    for student in sorted(reserve):
        left, right = student-1, student+1
        if left in lost:
            lost.remove(left)
        elif right in lost:
            lost.remove(right)
            
    return n - len(lost)